File: java/tests/integration/TestApp.java
package io.errordebugger.integration;

import io.errordebugger.Config;
import io.errordebugger.ErrorDebugger;

/**
 * Integration test app for Java SDK
 */
public class TestApp {
    public static void main(String[] args) throws InterruptedException {
        if (args.length < 2) {
            System.err.println("Usage: TestApp <dsn> <token>");
            System.exit(1);
        }
        
        String dsn = args[0];
        String token = args[1];
        
        // Initialize SDK
        Config config = new Config()
            .dsn(dsn)
            .token(token)
            .serviceName("java-integration-test")
            .debug(true);
            
        ErrorDebugger.init(config);
        
        // Trigger a test error
        Exception error = new RuntimeException("Integration test error from Java SDK");
        ErrorDebugger.captureException(error);
        
        // Allow time for export
        Thread.sleep(2000);
        ErrorDebugger.shutdown();
        System.out.println("Java SDK test completed");
    }
}

File: go/tests/integration/test_app.go
// Integration test app for Go SDK
package main

import (
	"context"
	"errors"
	"fmt"
	"os"
	"time"

	"errordebugger"
	"go.opentelemetry.io/otel/attribute"
)

func main() {
	if len(os.Args) < 3 {
		fmt.Println("Usage: test_app <dsn> <token>")
		os.Exit(1)
	}

	dsn := os.Args[1]
	token := os.Args[2]

	// Initialize SDK
	config := errordebugger.Config{
		DSN:         dsn,
		Token:       token,
		ServiceName: "go-integration-test",
		Debug:       true,
	}

	if err := errordebugger.Init(config); err != nil {
		fmt.Printf("Init failed: %v\n", err)
		os.Exit(1)
	}
	defer errordebugger.Shutdown()

	// Trigger a test error
	testErr := errors.New("Integration test error from Go SDK")
	errordebugger.CaptureException(
		context.Background(),
		testErr,
		attribute.String("test", "integration"),
		attribute.String("language", "go"),
	)

	// Allow time for export
	time.Sleep(2 * time.Second)
	fmt.Println("Go SDK test completed")
}
